src/logic/game.h

```diff
@@ -495,6 +502,7 @@ class Game : public EditorGameBase {
 
 	std::list<PendingDiplomacyAction> pending_diplomacy_actions_;
 	bool diplomacy_allowed_{true};
+	bool naval_warfare_allowed_{false};
```

Comment:
this needs to be initialised in true to make the BIT wise AND in line 411 of game.cc work 
this will make the checkbox work correctly in launch game.